src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1520,51 +1542,41 @@ private String quoteUnixCredentials(String str) {
+    private FilePath createWindowsSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTextTempFile("pass", ".bat",
+            "echo \"" + quoteWindowsCredentials(Secret.toString(sshUser.getPassphrase())) + "\"", false);
         return ssh;
     }
 
-    private File createUnixSshAskpass(SSHUserPrivateKey sshUser) throws IOException {
-        File ssh = File.createTempFile("pass", ".sh");
-        try (PrintWriter w = new PrintWriter(ssh)) {
-            w.println("#!/bin/sh");
-            w.println("echo '" + quoteUnixCredentials(Secret.toString(sshUser.getPassphrase())) + "'");
-        }
-        ssh.setExecutable(true);
+    private FilePath createUnixSshAskpass(SSHUserPrivateKey sshUser) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTextTempFile("pass", ".sh",
+              "#!/bin/sh\n"
+            + "echo \"" + quoteUnixCredentials(Secret.toString(sshUser.getPassphrase())) + "\"", false);
+        ssh.chmod(0700);
         return ssh;
     }
 
-    private File createWindowsStandardAskpass(StandardUsernamePasswordCredentials creds) throws IOException {
-        File askpass = File.createTempFile("pass", ".bat");
-        try (PrintWriter w = new PrintWriter(askpass)) {
-            w.println("@set arg=%~1");
-            w.println("@if (%arg:~0,8%)==(Username) echo " + quoteWindowsCredentials(creds.getUsername()));
-            w.println("@if (%arg:~0,8%)==(Password) echo " + quoteWindowsCredentials(Secret.toString(creds.getPassword())));
-        }
-        askpass.setExecutable(true);
+    private FilePath createWindowsStandardAskpass(StandardUsernamePasswordCredentials creds) throws IOException, InterruptedException {
+        FilePath askpass = workspace.createTextTempFile("pass", ".bat",
+                  "@set arg=%~1\r\n"
+                + "@if (%arg:~0,8%)==(Username) echo " + quoteWindowsCredentials(creds.getUsername()) + "\r\n"
+                + "@if (%arg:~0,8%)==(Password) echo " + quoteWindowsCredentials(Secret.toString(creds.getPassword())),
+                false);
         return askpass;
     }
 
-    private File createUnixStandardAskpass(StandardUsernamePasswordCredentials creds) throws IOException {
-        File askpass = File.createTempFile("pass", ".sh");
-        try (PrintWriter w = new PrintWriter(askpass)) {
-            w.println("#!/bin/sh");
-            w.println("case \"$1\" in");
-            w.println("Username*) echo '" + quoteUnixCredentials(creds.getUsername()) + "' ;;");
-            w.println("Password*) echo '" + quoteUnixCredentials(Secret.toString(creds.getPassword())) + "' ;;");
-            w.println("esac");
-        }
-        askpass.setExecutable(true);
+    private FilePath createUnixStandardAskpass(StandardUsernamePasswordCredentials creds) throws IOException, InterruptedException {
+        FilePath askpass = workspace.createTextTempFile("pass", ".sh",
```

Comment:
this smells insecure.  use [`WorkspaceList.tmpDir(workspace)`](https://github.com/jenkinsci/jenkins/commit/307bfc17385c14aefcb623ccc3e7144e4f01e2a3)

If you can not bump a version to that write a helper that does the same thing.